backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -269,98 +194,78 @@ public void addEmail(Email email) {
-                                .onItem().transformToMulti(Multi.createFrom()::iterable)
-                                .collect().in(() -> aggregator, AbstractEmailPayloadAggregator::aggregate).toMulti();
-                    }
+    private Uni<String> renderTitle(Notification item, EmailSubscriptionType emailSubscriptionType, EmailTemplate emailTemplate) {
+        return emailTemplate.getTitle(item.getAction().getEventType(), emailSubscriptionType)
+                .data("action", item.getAction())
+                .createMulti()
+                .collect().with(Collectors.joining())
+                .onFailure()
+                .recoverWithItem(templateEx -> {
+                    log.log(Level.WARNING, templateEx, () -> String.format(
+                            "Unable to render template title for application: [%s], eventType: [%s], subscriptionType: [%s].",
+                            item.getAction().getApplication(),
+                            item.getAction().getEventType(),
+                            emailSubscriptionType
+                    ));
+                    return null;
+                });
+    }
 
-                    if (delete) {
-                        // Nothing to do, delete them right away.
-                        return emailAggregationResources.purgeOldAggregation(aggregationKey, endTime)
-                                .toMulti()
-                                .onItem().transformToMultiAndMerge(i -> Multi.createFrom().empty());
+    private Uni<Tuple3<Email, String, String>> combine(Email email, Uni<String> title, Uni<String> body) {
+        return Uni.combine().all()
+                .unis(
+                        Uni.createFrom().item(email),
+                        title,
+                        body
+                ).asTuple()
+                .onItem().transform(objects -> {
+                    if (objects == null || objects.getItem1() == null || objects.getItem2() == null || objects.getItem3() == null) {
+                        return null;
                     }
 
-                    return Multi.createFrom().empty();
-                })
-                .onItem().transformToMultiAndConcatenate(aggregator -> {
-                    String accountId = aggregationKey.getAccountId();
-                    String bundle = aggregationKey.getBundle();
-                    String application = aggregationKey.getApplication();
-
-                    if (aggregator.getProcessedAggregations() == 0) {
-                        return Multi.createFrom().empty();
-                    }
+                    return objects;
+                });
+    }
 
-                    aggregator.setStartTime(startTime);
-                    aggregator.setEndTimeKey(endTime);
+    @Incoming(AGGREGATION_CHANNEL)
+    public Uni<Void> consumeEmailAggregations(String aggregationsJson) throws JsonProcessingException {
```

Comment:
Note for later: we'll need metrics in this method.

Reply:
I will create a Jira ticket for it so we don't lose this note